stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/common/SingleLiveEvent.kt

```diff
@@ -0,0 +1,45 @@
+package io.getstream.chat.ui.sample.common
+
+import androidx.annotation.MainThread
+import androidx.annotation.Nullable
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import io.getstream.chat.android.client.logger.ChatLogger
+import java.util.concurrent.atomic.AtomicBoolean
+
+class SingleLiveEvent<T> : MutableLiveData<T>() {
```

Comment:
This common `SingleLiveEvent` implementation has significant issues, there are a lot of things it doesn't handle correctly. It can't have multiple observers, it may lose values if multiple ones are posted when its observer isn't active, and removing observers won't work as expected either. As a bonus, the syntax for sending events is woeful.

I've written about this problem [here](https://zsmb.co/thoughts-about-event-handling-on-android/) and gave a talk about how I implemented this in my own framework [here](https://youtu.be/4U7hxKAZ7qE?t=1382). Let's discuss it, and see what our needs for events are exactly, and which of the problems around them we want to solve.